scripts/migrate_dimension.py

```
@@ -0,0 +1,161 @@
+                    await conn.execute("DELETE FROM vectors_metadata")
+                    await conn.execute(
+                        "INSERT INTO vectors_metadata (dimension) VALUES (?)", (current_dim,)
+                    )
+                    await conn.commit()
+                logger.info("vectors_metadata updated successfully.")
+
+                # Invalidate stale dimension cache in the storage adapter by recreating it.
+                await storage.dispose()
+                storage = await _create_storage_adapter(settings)
+                logger.info("Storage adapter re-initialized with new dimension.")
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)